block-producer/src/batch_builder/mod.rs

```diff
@@ -111,16 +112,23 @@ impl<BB> BatchBuilder for DefaultBatchBuilder<BB>
 where
     BB: BlockBuilder,
 {
+    #[allow(clippy::blocks_in_conditions)] // Workaround of `instrument` issue
+    #[instrument(target = "miden-block-producer", skip_all, err, fields(batch_id))]
     async fn build_batch(
         &self,
         txs: Vec<SharedProvenTx>,
     ) -> Result<(), BuildBatchError> {
         let num_txs = txs.len();
 
+        info!(target: COMPONENT, num_txs, "Building a transaction batch");
+        debug!(target: COMPONENT, txs = %format_array(txs.iter().map(|tx| tx.id().to_hex())));
+
         let batch = Arc::new(TransactionBatch::new(txs)?);
-        self.ready_batches.write().await.push(batch);
 
-        info!(COMPONENT, "batch built with {num_txs} txs");
+        info!(target: COMPONENT, "Transaction batch built");
+        Span::current().record("batch_id", format_blake3_digest(batch.id()));
+
+        self.ready_batches.write().await.push(batch);
 
```

Comment:
Let's add another info event here: "Transaction batch added to the batch queue" and add a filed for this event with the total number of batches in the queue.